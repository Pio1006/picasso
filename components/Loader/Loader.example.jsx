import React from 'react'
import { storiesOf } from '@storybook/react'

import StoryTeller from '../../.storybook/StoryTeller'
import Loader from './'
import Spacer from '../Spacer'

const stories = storiesOf('Loader', module)

const teller = new StoryTeller('Loader', 'indicate progress')
const chapter = teller.addChapter()

chapter
  .addSection('Default', 'Default loader', () => <Loader />)
  .addSection('Labeled', 'Loader with label', () => (
    <Loader label='Loading...' />
  ))
  .addSection('Indeterminate', 'Indeterminate loader', () => (
    <Loader indeterminate label='Loading...' />
  ))
  .addSection('Sizes', 'Different size loader', () => (
    <div>
      <Loader label='small' size='small' />
      <Spacer bottom={2} />
      <Loader label='default' size='default' />
      <Spacer bottom={2} />
      <Loader label='large' size='large' />
    </div>
  ))
  .addSection('Inline', 'Place loader inline with content', () => (
    <div style={{ width: '100%' }}>
      <Loader inline />
      <span>Content...</span>
    </div>
  ))
  .addSection('Value', 'Loader with determined or static values', () => (
    <div>
      <Loader label='50%' value={50} variant='static' />
      <Spacer bottom={2} />
      <Loader label='13%' value={13} variant='determinate' />
    </div>
  ))

stories.addWithChapters('Loader', teller.toStory())
