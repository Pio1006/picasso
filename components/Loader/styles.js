export default () => ({
  spinner: {
    color: 'inherit'
  },
  wrapper: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  label: {
    marginTop: '1rem'
  },
  inline: {
    display: 'inline-flex'
  }
})
