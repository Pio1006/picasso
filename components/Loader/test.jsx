import React from 'react'
import { render, cleanup } from 'react-testing-library'

import Loader from './index'

const renderLoader = (children, props = {}) => {
  return render(<Loader {...props}>{children}</Loader>)
}

afterEach(cleanup)

test('props combo', () => {
  const { container } = renderLoader({
    inline: true,
    label: 'Testing',
    size: 'large',
    indeterminate: true
  })

  expect(container).toMatchSnapshot()
})
