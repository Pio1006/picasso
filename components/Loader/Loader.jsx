import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import cx from 'classnames'

import CircularProgress from '../CircularProgress'
import styles from './styles'

const SIZES = {
  small: 20,
  default: 40,
  large: 80
}

const getSize = size => {
  return SIZES[size] || SIZES.default
}

const Loader = props => {
  const { label, classes, variant, size, inline, className, value } = props

  return (
    <div
      className={cx(classes.wrapper, className, {
        [classes.inline]: inline
      })}
    >
      <CircularProgress
        classes={{
          root: classes.spinner
        }}
        size={getSize(size)}
        value={value}
        variant={variant}
      />

      {label && <div className={classes.label}>{label}</div>}
    </div>
  )
}

Loader.propTypes = {
  label: PropTypes.string,
  variant: PropTypes.oneOf(['determinate', 'indeterminate', 'static']),
  size: PropTypes.oneOf(['small', 'default', 'large']),
  inline: PropTypes.bool,
  classes: PropTypes.shape({
    root: PropTypes.string
  }).isRequired,
  className: PropTypes.string,
  value: PropTypes.number
}

Loader.defaultProps = {
  className: '',
  inline: false,
  label: null,
  size: 'default',
  value: 0,
  variant: 'indeterminate'
}

export default withStyles(styles)(Loader)
