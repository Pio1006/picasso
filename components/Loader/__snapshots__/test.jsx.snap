// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props combo 1`] = `
<div>
  <div
    class="Loader-wrapper"
  >
    <div
      class="MuiCircularProgress-root CircularProgress-root Loader-spinner MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg-11"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle-12 MuiCircularProgress-circleIndeterminate-14"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
